### PR TITLE
Check requirements URLs before building

### PR DESCRIPTION
**Pull request by jrichter1**
To avoid certain situations that have arisen throughout the development, this PR aims to check the requirements before building so the urls:
 - respond with 200
 - point to a file with roughly the size we expect
 - contain some text specific to the tool that should be downloaded (i.e. vagrant url contains 'vagrant' - we've been there before :) )

### CODE REVIEW

File: test/check-requirements.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+let reqs = require('../requirements.json');
+let request = require("request");
+
+let minSizes = new Object();
```
Comment (ppitonak) [inline test/check-requirements.js]: why don't you use simply `let minSizes = {};`?

File: test/check-requirements.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  if (!data[key].includes(fileNames[key])) {
+    throw new Error(key + ' - the url does not contain ' + fileNames[key] + ': ' + data[key]);
+  }
+}
+
+function checkUrl(key) {
+  let req = request.get(data[key])
+  .on('response', function(response) {
+    let size = response.headers['content-length'];
+    if (response.statusCode !== 200) {
```
Comment (ppitonak) [inline test/check-requirements.js]: What happens if the URL will redirect to a new place?

Comment (jrichter1) [inline test/check-requirements.js]: Both virtualbox and vagrant use redirects, something like this happens:
virtualbox.exe - url: http://developers.redhat.com/redirect/to/virtualbox-5.0.8.download
virtualbox.exe - size: 117095112B

**Comment by ppitonak** [comment]
Merged

**Pull request closed without merging**